src/rustup.ts

```diff
@@ -172,4 +172,27 @@ async function installRls(): Promise<void> {
     }
 
     stopSpinner('RLS components installed successfully');
-}
\ No newline at end of file
+}
+
+/**
+ * Parses given output of `rustup show` and retrieves the local active toolchain.
+ */
+export function parseActiveToolchain(rustupOutput: string): string {
```

Comment:
This isn't `async`, since it needs to finish to return a value with which global [`RLSConfiguration`](https://github.com/rust-lang-nursery/rls-vscode/pull/220/files#diff-8b01d5598b81332eb62385d26f3ead22R64) is created.
For the same reason it only parses the result and it's not a complete function that uses `CONFIGURATION.rustupPath` value, since it's called during `CONFIGURATION` construction.

As a side note, the configuration probably will need to be redesigned later on, as currently it's a global that's lazily constructed once and doesn't handle `'rust-client.*'` value changes

Reply:
Could you add the 'side note' as a FIXME or file an issue please?